web/src/echo.ts

```diff
@@ -119,24 +212,8 @@ export function build_display_recipient(message) {
     const display_recipient = emails.map((email) => {
         email = email.trim();
         const person = people.get_by_email(email);
-        if (person === undefined) {
-            // For unknown users, we return a skeleton object.
-            //
-            // This allows us to support zephyr mirroring situations
-            // where the server might dynamically create users in
-            // response to messages being sent to their email address.
-            //
-            // TODO: It might be cleaner for the web app for such
-            // dynamic user creation to happen inside a separate API
-            // call when the pill is constructed, and then enforcing
-            // the requirement that we have an actual user object in
-            // `people.js` when sending messages.
-            return {
-                email,
-                full_name: email,
-                unknown_local_echo_user: true,
-            };
-        }
```

Comment:
If you are removing this then type `DisplayRecipentUser` will never contain `unknown_local_echo_user`. In that case, we should drop that completely. It is just used in one place.

Reply:
Sure. https://github.com/zulip/zulip/pull/30702/commits/738b93517d86566f3bead2aef7e98ac501188eaa drops all instances of `unknown_local_echo_user` from frontend.